sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -492,7 +537,7 @@ func (n *ProcessSizedElementsAndRestrictions) Split(f float64) ([]*FullValue, []
 // behavior is identical). A single restriction split will occur and all windows
 // present in the unsplit element will be present in both the resulting primary
 // and residual.
-func (n *ProcessSizedElementsAndRestrictions) singleWindowSplit(f float64) ([]*FullValue, []*FullValue, error) {
+func (n *ProcessSizedElementsAndRestrictions) singleWindowSplit(f float64, pWeState interface{}, rWeState interface{}) ([]*FullValue, []*FullValue, error) {
```

Comment:
Nit here and similar methods below: You can avoid repeating the type for each input parameter of an identical type. eg `a int, b int, c int, d int` => `a, b, c, d int` 

This also works for identical fields in a struct declaration.

```suggestion
func (n *ProcessSizedElementsAndRestrictions) singleWindowSplit(f float64, pWeState, rWeState interface{}) ([]*FullValue, []*FullValue, error) {
```